dotnet/src/Connectors/Connectors.Memory.DuckDB/DuckDBMemoryStore.cs

```diff
@@ -152,22 +150,16 @@ public async Task RemoveBatchAsync(string collectionName, IEnumerable<string> ke
             yield break;
         }
 
-        var collectionMemories = new List<MemoryRecord>();
         TopNCollection<MemoryRecord> embeddings = new(limit);
 
-        await foreach (var record in this.GetAllAsync(collectionName, cancellationToken))
+        await foreach (var dbEntry in this._dbConnector.GetNearestMatchesAsync(this._dbConnection, collectionName, embedding.ToArray(), limit, minRelevanceScore, cancellationToken))
```

Comment:
I think you can delete `GetAllAsync` now too, right?

Reply:
let me look!